src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileReader.kt

```diff
@@ -59,9 +59,13 @@ class CsvFileReader internal constructor(
      * read all csv rows as Sequence with header information
      */
     fun readAllWithHeaderAsSequence(): Sequence<Map<String, String>> {
-        val headers = readNext() ?: return emptySequence()
-        val duplicated = findDuplicate(headers)
-        if (duplicated != null) throw MalformedCSVException("header '$duplicated' is duplicated")
+        var headers = readNext() ?: return emptySequence()
+        if (ctx.autoRenameDuplicateHeaders) {
+            headers = deduplicateHeaders(headers)
+        } else {
+            val duplicated = findDuplicate(headers)
+            if (duplicated != null) throw MalformedCSVException("header '$duplicated' is duplicated")
```

Comment:
It would be nice to include a hint in the MalformedCSVException message, which indicates the `autoRenameDuplicateHeaders` option can avoid Exception.